ntoskrnl/inbv/inbv.c

```diff
@@ -793,13 +792,29 @@ InbvRotationThread(
     }
     X = ProgressBarLeft + 2;
     Y = ProgressBarTop + 2;
+    InbvReleaseLock();
 
-    while (InbvDisplayState == INBV_DISPLAY_STATE_OWNED && PltRotBarStatus != 3)
+    while (InbvDisplayState == INBV_DISPLAY_STATE_OWNED)
     {
-        RotBarType = RotBarSelection;
-        InbvReleaseLock();
+        InbvAcquireLock();
```

Comment:
Actually the drawing with VidXXX() should be done under Inbv lock, and since also the PltRotBarStatus should be checked under lock I moved this one here. InbvDisplayState can be however checked outside lock.

Reply:
Also now RotBarSelection checkings are done inside Inbv lock so the temporary variable is unneeded here.

Reply:
Lgtm